nova/cmd/manage.py

```diff
@@ -2453,14 +2454,174 @@ def sync_aggregates(self, verbose=False):
+            if not flavor or not _should_update(instance, flavor):
+                return
+            if dry_run:
+                output("Instance %s has outdated flavor extra_specs."
+                       % instance.uuid)
+                return
+            instance.flavor.extra_specs = flavor.extra_specs
+            instance.save()
+            _update_request_spec_flavor(instance, flavor)
+
+            output("Updated flavor extra_specs for instance %s. "
+                   % instance.uuid)
+
+        self._run_across_cells(_update_instance_flavor,
+                               instance_uuid=instance_uuid,
+                               expected_attrs=['flavor'],
+                               max_count=max_count,
+                               dry_run=dry_run,
+                               output=output)
+
+    def _run_across_cells(self, cb, instance_uuid=None,
+                          expected_attrs=None, filters=None,
+                          max_count=50, dry_run=False,
+                          output=print):
+
+        ctxt = context.get_admin_context()
+
+        if instance_uuid:
+            try:
+                im = objects.InstanceMapping.get_by_instance_uuid(
+                    ctxt, instance_uuid)
+                cells = objects.CellMappingList(objects=[im.cell_mapping])
+            except exception.InstanceMappingNotFound:
+                print('Unable to find cell for instance %s, is it mapped? Try '
+                      'running "nova-manage cell_v2 verify_instance" or '
+                      '"nova-manage cell_v2 map_instances".' %
+                      instance_uuid)
+                return 127
+        else:
+            cells = objects.CellMappingList.get_all(ctxt)
+            if not cells:
+                output(_('No cells to process.'))
+                return 4
+
+        for cell in cells:
+            if cell.uuid == objects.CellMapping.CELL0_UUID:
+                continue
+            output(_('Looking for instances in cell: %s') % cell.identity)
+
+            with context.target_cell(ctxt, cell) as cctxt:
+                self._run_on_instances_batch(
+                    cctxt, cb, instance_uuid=instance_uuid,
+                    expected_attrs=expected_attrs, max_count=max_count,
+                    filters=filters, dry_run=dry_run, output=output)
+
+    def _run_on_instances_batch(self, ctxt, cb, instance_uuid=None,
+                                expected_attrs=None, max_count=50,
+                                filters=None, dry_run=False, output=print):
+        if instance_uuid:
+            max_count = 1
+            unlimited = False
+        elif max_count is not None:
+            try:
+                max_count = int(max_count)
+            except ValueError:
+                max_count = -1
+            unlimited = False
+            if max_count < 1:
+                print(_('Must supply a positive integer for --max-count.'))
+                return 127
```

Comment:
This does not really bubble up anywhere, because `_run_across_cells()` does not handle our return code. Therefore, returning 127 seems mood.

Additionally, with returning `num_processed`, we might return `127` in non-error-cases, too.